### ISSUE TITLE
Install request: Voronoi for LAMMPS

### ISSUE THREAD

**Issue by heatherkellyucl** [UCL-RITS/rcps-buildscripts#199]
IN:03143536
https://lammps.sandia.gov/doc/Packages_details.html#voronoi-package

Requires Voro++ which is an external library. The lammps build system can download and build it as part of the process.
```bash
make lib-voronoi                          # print help message
make lib-voronoi args="-b"                # download and build the default version in lib/voronoi/voro++-<version>
make lib-voronoi args="-p $HOME/voro++"   # use existing Voro++ installation in $HOME/voro++
make lib-voronoi args="-b -v voro++0.4.6" # download and build the 0.4.6 version in lib/voronoi/voro++-0.4.6
```

**Comment by heatherkellyucl** [comment]
It would be useful to also have most of the non-user components built.

These ought to be simple:
`asphere body class2 colloid compress coreshell dipole granular kspace manybody mc misc molecule mpiio peri qeq replica rigid shock snap srd`

These require external libs and have similar ways to build them:
`kim latte meam mscg poems reax voronoi`

The others are optimisation and we already have the `gpu` and `user-intel` versions for those purposes in the most recent build.

**Comment by owainkenwayucl** [comment]
Right, I'm doing some testing on myriad.

The problem is, the feature in LAMMPS which autobuilds a library for you does zero verification of the file it's downloaded, and for voronoi (not checked others) its a bare HTTP request.  Since this feature also depends on a more modern version of python than the RHEL7 default I guess this means we have to download and build the library as part of the LAMMPS install ourselves.

Ahead of this I'm doing a test build with *all* the packages that don't rely on external libs.

**Comment by owainkenwayucl** [comment]
Right.  The build with all internal packages seems to have worked.  I'll make a start on the ones with external libs tomorrow.

**Comment by owainkenwayucl** [comment]
Both meam and reax generate errors when I try to use `install.py` to compile them.

**Comment by owainkenwayucl** [comment]
Right, the problem with meam/reax is that they try to build with `mpifort` which of course doesn't exist, possibly anywhere in the world.

**Comment by owainkenwayucl** [comment]
OK, looking at LATTE, in future releases LAMMPS will checksum it but at the moment lammps stable (16Mar18) doesn't.

But it is downloading over https: from github so...

**Comment by owainkenwayucl** [comment]
(meam + reax also require that you add -lifcore to your link line)

**Comment by owainkenwayucl** [comment]
Right, I have it building with `voronoi`.  Just `kim`, `mcsg` and `latte` to do.

**Comment by owainkenwayucl** [comment]
Looking at kim, mcsg and latte I'm going to skip those for reasons.  MSCG adds GSL as a dependency to LAMMPS, latte looks like a real hassle to build and kim is itself a big complicated framework.

**Comment by owainkenwayucl** [comment]
Right, the CPU versions are deployed on each cluster, waiting for GPU nodes to become free so I can build those on Legion/Myriad.

**Comment by owainkenwayucl** [comment]
I've contacted the user.
